devito/compiler.py

```diff
@@ -12,7 +12,20 @@
 
 
 class Compiler(GCCToolchain):
-    """Base class for all compiler classes."""
+    """Base class for all compiler classes.
+
+    :param openmp: boolean indicating if openmp is enabled. False by default
+    :param cc: name of compiler used. Set to unknown.
```

Comment:
Ok, so I don't think all of theses `param`s (except for `openmp`) need to go here, since they are not part of the actual constructor signature and therefore not user-facing. That being said, a nice informative paragraph about how to sub-class the base compiler might be helpful, something like:
```
The base class defaults all compiler specific settings to empty lists. Preset configurations
can be built by inheriting from `Compiler` and setting specific flags to the desired values, eg.:

def class MyCompiler(Compiler):
    def __init__(self):
        self.cc = 'mycompiler'
        self.cflags = ['list', 'of', 'all', 'compiler', 'flags']
```